File: client/classi/BuonoDiConsegna.js
class BuonoDiConsegna {
    constructor(id, idPolizza, idCliente, pesoRiferito, dataEmissione) {
        this.id = id;
        this.idPolizza = idPolizza;
        this.idCliente = idCliente;
        this.pesoRiferito = pesoRiferito;
        this.dataEmissione = dataEmissione; // YYYY-MM-DD
    }

    static fromJSON(json) {
        return new BuonoDiConsegna(
            json.id,
            json.idPolizza,
            json.idCliente,
            json.pesoRiferito,
            json.dataEmissione
        );
    }

    toJSON() {
        return {
            id: this.id,
            idPolizza: this.idPolizza,
            idCliente: this.idCliente,
            pesoRiferito: this.pesoRiferito,
            dataEmissione: this.dataEmissione
        };
    }
}

File: server/demo/src/main/java/com/example/demo/Model/BuonoDiConsegna.java
package com.example.demo.Model;

public class BuonoDiConsegna {
    private int id;
    private int id_polizza;
    private int id_cliente;
    private double peso_riferito;
    private String data_emissione; // data come stringa

    // Costruttore vuoto
    public BuonoDiConsegna() {}

    // Costruttore completo
    public BuonoDiConsegna(int id, int id_polizza, int id_cliente, double peso_riferito, String data_emissione) {
        this.id = id;
        this.id_polizza = id_polizza;
        this.id_cliente = id_cliente;
        this.peso_riferito = peso_riferito;
        this.data_emissione = data_emissione;
    }

    // Getter e Setter
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId_polizza() {
        return id_polizza;
    }

    public void setId_polizza(int id_polizza) {
        this.id_polizza = id_polizza;
    }

    public int getId_cliente() {
        return id_cliente;
    }

    public void setId_cliente(int id_cliente) {
        this.id_cliente = id_cliente;
    }

    public double getPeso_riferito() {
        return peso_riferito;
    }

    public void setPeso_riferito(double peso_riferito) {
        this.peso_riferito = peso_riferito;
    }

    public String getData_emissione() {
        return data_emissione;
    }

    public void setData_emissione(String data_emissione) {
        this.data_emissione = data_emissione;
    }
}
